englishLearnBot-stage5

Name707 opened the pull request:
Реализован функционал:
1. Считывать ответ пользователя и сравнивать с правильным ответом по индексу списка слов для вопросов.

2. Создать функцию сохранения обновленных результатов в словарь и вызвать ее при каждом правильном ответе.

3. В вариантах ввода добавить новый пункт (0): Меню (направляет пользователя в главное меню).

Name707 committed:
englishLearnBot-stage5
Реализован функционал:
1. Считывать ответ пользователя и сравнивать с правильным ответом по индексу списка слов для вопросов.

2. Создать функцию сохранения обновленных результатов в словарь и вызвать ее при каждом правильном ответе.

3. В вариантах ввода добавить новый пункт (0): Меню (направляет пользователя в главное меню).

ievetrov commented on src/main/kotlin/Main.kt:
```diff
@@ -51,7 +62,7 @@ fun main() {
+    var correctAnswersCount: Int = 0,
 ) {
 
 }
```
Тут не нужны скобки, у нас в Word не будет методов, это класс для хранения данных

ievetrov commented on src/main/kotlin/Main.kt:
```diff
@@ -23,17 +28,23 @@ fun main() {
                     break
                 } else {
-                    println("Выберите правильный перевод слова \"${randomFourUnlearnedWords.random().original}\":\n" +
+                    println("Введите цифру правильного варианта перевода слова \"${randomWordForLearn?.original}\". Цифра \"0\" для выхода в меню:\n" +
```
Мы в требованиях к тренажеру описывали формат запроса слов:

> hello
> 1 - зеленый, 2 - кот, 3 - здравствуйте, 4 - собака

Кажется все таки понятнее когда мы просим ввести конкретную цифру, иначе есть риск недопонимания.
https://youtu.be/DNZoVTdM3uQ?t=107

ievetrov commented on src/main/kotlin/Main.kt:
```diff
@@ -23,17 +28,23 @@ fun main() {
+                    val userAnswer = readln().toInt()
 
+                    if (userAnswer != 0 && (randomFourUnlearnedWords[userAnswer - 1].translate == randomWordForLearn?.translate)) {
+                        dictionary.map { if (it.translate == randomWordForLearn.translate) it.correctAnswersCount++ }
```
Нам не обязательно искать слово в словаре, это к тому же будет толще происходить, так как нужно пробежать весь список.
Мы можем прям взять:
```Kotlin
val userAnswer = randomFourUnlearnedWords[userAnswerIndex - 1]
if (userAnswerIndex != 0 && userAnswer.translate == randomWordForLearn?.translate) {
    userAnswer.correctAnswersCount++ // так как объекты в Kotlin хранятся по ссылке, значение также изменится и в Dictionary, а следовательно и сохранится
    saveDictionary()
} else {
    // ...
}
```

Name707 committed:
englishLearnBot-stage5
Исправление замечаний. Формат вывода вариантов ответа (1 - вариант1, 2 - вариант2...). У класса Word() удален вторичный конструктор
